packages/auth/amplify_auth_cognito_dart/lib/src/state/machines/fetch_auth_session_state_machine.dart

```diff
@@ -239,22 +265,45 @@ class FetchAuthSessionStateMachine extends FetchAuthSessionStateMachineBase {
       );
     }
 
-    final awsCredentialsResult = await _retrieveAwsCredentials(
-      existingIdentityId: event.request.options?.developerProvidedIdentityId ??
-          result.data.identityId,
-      federatedIdentity: _FederatedIdentity(
-        event.request.provider,
-        event.request.token,
-      ),
-    );
+    AuthResult<AWSCredentials> credentialsResult;
+    AuthResult<String> identityIdResult;
+
+    try {
+      final res = await _retrieveAwsCredentials(
+        existingIdentityId:
+            event.request.options?.developerProvidedIdentityId ??
+                result.data.identityId,
+        federatedIdentity: _FederatedIdentity(
+          event.request.provider,
+          event.request.token,
+        ),
+      );
+      credentialsResult = AuthResult.success(res.awsCredentials);
+      identityIdResult = AuthResult.success(res.identityId);
+    } on AuthException catch (e) {
+      credentialsResult = AuthResult.error(e);
+      identityIdResult = AuthResult.error(e);
+    } on Object catch (e) {
+      final authException = UnknownException(
+        'An unknown exception occurred.',
+        underlyingException: e,
+      );
+      credentialsResult = AuthResult.error(authException);
+      identityIdResult = AuthResult.error(authException);
+    }
```

Comment:
Catching `Object` is risky in general since we may swallow errors that are not expected, like incorrect code. At the state machine layer and above, I'm not sure it's every really needed.

```suggestion
    } on Exception catch (e) {
      final authException = AuthException.fromException(e);
      credentialsResult = AuthResult.error(authException);
      identityIdResult = AuthResult.error(authException);
    }
```

Reply:
+1